site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -47,7 +47,87 @@ private function validateDiffViewerType(string $type) {
         }
         return true;
     }
-
+    /* Used in ajaxRandomizePeets as a helper function.
+    @param  $type
+    */
+    public function arr_rotate($array,$rotate_count) {
+        
+    }
+/**
+        * Route for randomizing peer assignments
+        * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/ajaxRandomizePeers", methods={"POST"})
+        * @AccessControl(role="INSTRUCTOR")
+    */ 
+    public function ajaxRandomizePeers($gradeable_id){
+        $number_to_grade=$_POST['number_to_grade'];
+       // $gradeable_id=$_POST['gradeable_id'];
+        $gradeable = $this->tryGetGradeable($gradeable_id);
+         if ($gradeable === false) {
+             $this->core->addErrorMessage('Invalid Gradeable!');
+             $this->core->redirect($this->core->buildCourseUrl());
+         }
+         $order = new GradingOrder($this->core, $gradeable, $this->core->getUser(), true);
+         $student_array= [];
+         $student_list= [];
+         $students = $this->core->getQueries()->getUsersByRegistrationSections($order->getSectionNames());
+         foreach ($students as $student) {
+              $reg_sec = ($student->getRegistrationSection() === null) ? 'NULL' : $student->getRegistrationSection();
+              $sorted_students[$reg_sec][] = $student;
+              array_push($student_list, [
+                  'user_id' => $student->getId(),
+                 // reg_section => $reg_sec,
+              ]);
+              array_push($student_array,$student->getId());
+          }
+          $graded_array=$student_array;
+          $final_info=[];
+          shuffle($student_array);
+          array_push($final_info,$student_array);
+          $final_grading_info=[];
```

Comment:
There is a `final_info` and a `final_grading_info` here. Can we name them more descriptively so that I can tell the difference?

Reply:
Renamed!